### ISSUE TITLE
Handling of value classes

### ISSUE THREAD

**Issue by sebastianvoss** [softwaremill/tapir#71]
I'm using value classes to have a type-safe representation of e.g. IDs.

For example, I have this case class:

`case class UserId(value: String) extends AnyVal`

With the help of `io.circe.generic.extras.deriveUnwrappedEncoder` this is represented in JSON as `"id": "1"` instead of `"id": {"value": "1"}`.

How can I make tapir aware of this situation. It currently outputs this OpenAPI doc:

```
id:
  required:
  - value
  type: object
  properties:
    value:
      type: string
```
instead of:
```
id:
  type: string
```

**Comment by adamw** [comment]
That would be somewhere in the schema derivation. Most probably `SchemaForMagnoliaDerivation`, where the schema for case classes is derived. Maybe there's a way to check in Magnolia if the  class is a value class?

**Comment by sebastianvoss** [comment]
Thanks @adamw. I gave it a try in https://github.com/softwaremill/tapir/pull/72. Would be great if you could have a look and let me know if this is the right approach.

**Comment by sebastianvoss** [comment]
Implemented in #72

**Comment by tobiajo** [comment]
Was this reverted? Tapir does the opposite for me now.
